src/pages/resultsView/download/CaseAlterationTable.tsx

```diff
@@ -0,0 +1,141 @@
+import {observer} from "mobx-react";
+import * as React from 'react';
+import LazyMobXTable from "shared/components/lazyMobXTable/LazyMobXTable";
+import {OQLLineFilterOutput} from "shared/lib/oql/oqlfilter";
+import {AnnotatedExtendedAlteration} from "../ResultsViewPageStore";
+
+export interface ISubAlteration {
+    type: string;
+    value: string;
+}
+
+export interface IOqlData {
+    geneSymbol: string;
+    mutation: string[];
+    fusion: string[];
+    cna: ISubAlteration[];
+    mrnaExp: ISubAlteration[];
+    proteinLevel: ISubAlteration[];
+}
+
+export interface ICaseAlteration {
+    studyId: string;
+    sampleId: string;
+    patientId: string;
+    altered: boolean;
+    oqlData: {[oqlLine: string]: IOqlData};
+}
+
+export interface ICaseAlterationTableProps {
+    caseAlterationData: ICaseAlteration[];
+    oqls: OQLLineFilterOutput<AnnotatedExtendedAlteration>[];
+}
+
+export function generateOqlValue(data: IOqlData): string
+{
+    const oqlValue: string[] = [];
+
+    if (data.mutation.length > 0) {
```

Comment:
lets write this function with a table?
{  test:xxxx, value: "MUT"  }

Reply:
@alisman updated `generateOqlValue` function, now using a helper object for each alteration type:
```JavaScript
{
    label: "MUT",
    getAlterationData: (oqlData: IOqlData) => oqlData.mutation,
    getValues: (alterationData: (string|ISubAlteration)[]) => alterationData
}
```